### PR TITLE
Fixes to MPI library

### PR DESCRIPTION
**Pull request by hanno-arm**
This PR suggests two fixes to the MPI library: 

  1. It corrects the sign computation for modular exponentiation as well as the relevant test case. 
  2. It adds a check aborting modular inversion in `mbedtls_mpi_inv_mod` when the modulus is `1`, which doesn't make sense. Previously, this case led to the routine hanging as noted in https://github.com/ARMmbed/mbedtls/issues/641; there doesn't seem to be a problem with `mpi_sub_help`, though.

Please review.

__Backports:__ Needs to be backported to mbed TLS 1.3 and 2.1.

### CODE REVIEW

**Comment by hanno-arm** [comment]
@gilles-peskine-arm Yes it does, that's deliberate at the moment, but I was also wondering whether we'd want change this at some point.

File: include/mbedtls/bignum.h [hunk 1]
```diff
@@ -615,13 +615,13 @@ int mbedtls_mpi_mod_int( mbedtls_mpi_uint *r, const mbedtls_mpi *A, mbedtls_mpi_
  * \param X        Destination MPI
  * \param A        Left-hand MPI
  * \param E        Exponent MPI
- * \param N        Modular MPI
+ * \param N        Modular MPI, must be odd
```
Comment (hanno-arm) [inline include/mbedtls/bignum.h]: @yanesca I'm afraight I don't understand what you mean. Where do you want this restriction on the parameter be documented instead?

**Comment by hanno-arm** [comment]
@yanesca I've removed most of the PR and reduced it to it's minimum related to what's mentioned in the ChangeLog.

File: ChangeLog [hunk 2]
```diff
@@ -1,5 +1,13 @@
 mbed TLS ChangeLog (Sorted per branch, date)
 
+= mbed TLS x.x.x branch released xxxx-xx-xx
+
+Bugfix
+   * Fix modular inversion function on invalid modulus 1.
+     Found by blaufish. Fixes #641.
+   * Fix sign bug in modular exponentiation.
+     Found by Guido Vranken.
```
Comment (andresag01) [inline ChangeLog]: Could you please describe a little bit more what the "sign bug" problem is?

**Comment by hanno-arm** [comment]
@andresag01 @yanesca @gilles-peskine-arm Could you also have a look at the backports https://github.com/ARMmbed/mbedtls/pull/909 and https://github.com/ARMmbed/mbedtls/pull/910 here? Thanks!

**Pull request merged**